system/CodeIgniter.php

```diff
@@ -1088,6 +1087,8 @@ protected function sendResponse()
      * without actually stopping script execution.
      *
      * @param int $code
+     *
+     * @deprecated 4.4.0 No longer Used. Moved to index.php.
```

Comment:
I am thinking that it actually fine to have `callExit()` called in `run()` method, the usage can be on Mocking CodeIgniter class as a service

Reply:
I don't think `CodeIgniter` should call `exit()`.
If `CodeIgniter` does not call `exit()`, there is no need to mock the method.

Reply:
I like this. In order to extend `CodeIgniter` I always have to disable this in projects:
```php
    /**
     * Workaround for testing so this class can be used
     * instead of the typical MockCodeIgniter.
     *
     * @param mixed $code
     */
    protected function callExit($code)
    {
        if (ENVIRONMENT !== 'testing') {
            parent::callExit($code);
        }
    }
```
Moving the exit call takes this out of the class concern.

Reply:
@samsonasik `CodeIgniter` should not call `exit()`.

Also, `spark` calls `exit(()`:
https://github.com/codeigniter4/CodeIgniter4/blob/4bc7a02a7153a69fc02bdc3f4d6ba75c36c77b15/spark#L99
This PR changes `index.php` to a structure similar to `spark`.

Reply:
Ok 👍